frontend/lib/src/components/transaction_component/transaction_component.dart

```diff
@@ -0,0 +1,159 @@
+import 'package:flutter/material.dart';
+import 'package:frontend/src/components/button.dart';
+import 'package:get/get.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:form_builder_validators/form_builder_validators.dart';
+
+import 'package:frontend/src/components/input/input_field.dart';
+
+/*
+  Reusable transaction component for deposit, withdraw, and transfer cash
+
+  @param label: String value to display what kind of transaction it is,
+    basically something like a title.
+
+  @param type: String value representing the type of transaction: DEPOSIT, WITHDRAWAL, or TRANSFER.
+    Determines the fields to display and the validation logic.
+*/
+
+class TransactionComponent extends StatefulWidget {
+  const TransactionComponent({
+    Key? key,
+    required this.label,
+    this.type = 'DEPOSIT',
+  }) : super(key: key);
+
+  final String label;
+  final String type;
+
+  @override
+  State<TransactionComponent> createState() {
+    return _TransactionComponentState();
+  }
+}
+
+class _TransactionComponentState extends State<TransactionComponent> {
+  final TextEditingController amountController = TextEditingController();
+  final TextEditingController emailController = TextEditingController();
+
+  void submitAmountData() {
+    final enteredAmount = double.tryParse(amountController.text);
+    final amountIsInvalid = enteredAmount == null || enteredAmount <= 0;
+    final emailIsInvalid = widget.type == 'TRANSFER' &&
+        (!emailController.text.trim().isEmail ||
+            emailController.text.trim().isEmpty);
```

Comment:
this is validating the inputs after the user has clicked the button, which i think is a bad practice since we have a way to validate the inputs before the user submits, or in other words, while the user is typing the inputs

the function is for submitting the form already but it includes the validation logic as well

Reply:
@jstephend-sun this is for showing the AlertDialog sir if the user click the confirm button without entering the correct email or amount or if the user click the confirm button without fillouting the field